### PR TITLE
Remove generated config.proto, switch to ghodss/yaml

### PR DESCRIPTION
**Pull request by fejta**
The `ghodss/yaml` library allows us to reuse the generated json tags.
The `go_proto_library()` rule allows us to generate the library from the `.proto` at compile time.

I'm going to restructure the files a little bit next.

/assign @krzyzacy

### CODE REVIEW

File: testgrid/config/README.md [hunk 1]
```diff
@@ -345,23 +345,3 @@ go build ./yaml2proto  # Build the yaml2proto library
...
-3. Search-replace all json:"foo,omitempty" with yaml:"foo,omitempty".
-```
-  # Be sure to add back the header
-  sed -i -e 's/json:/yaml:/g' pb/config.pb.go
-```
-4. Commit both `config.proto` and `config.pb.go`
-
-
-[`config.proto`]: ./config.proto
-[`config.yaml`]: ./config.yaml
```
Comment (krzyzacy) [inline testgrid/config/README.md]: keep it :-) this is also referenced elsewhere in the doc

**Comment by fejta** [comment]
The proto still generates json tags, but the new yaml library knows how to convert the json tags into json. This is how prow works [example](https://github.com/kubernetes/test-infra/blob/master/prow/config/config.go#L40)

File: testgrid/config/config_test.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ import (
 
 	"gopkg.in/yaml.v2"
```
Comment (krzyzacy) [inline testgrid/config/config_test.go]: also, here we can just switch to ghodss/yaml as well and drop gopkg.in/yaml dep?

Comment (fejta) [reply testgrid/config/config_test.go]: Yes and no.

Yes we can switch to that, but ghodss is just a wrapper for the same thing: https://github.com/ghodss/yaml/blob/master/yaml.go#L10

Comment (krzyzacy) [reply testgrid/config/config_test.go]: ah right 👍 , forgot about that part

Comment (fejta) [reply testgrid/config/config_test.go]: Done

**Pull request merged**